remote.go

```diff
@@ -864,6 +865,21 @@ func getHavesFromRef(
 	toVisit := maxHavesToVisitPerRef
 	return walker.ForEach(func(c *object.Commit) error {
 		haves[c.Hash] = true
+
+		// initialise the shallows map
+		if *shallows == nil {
```

Comment:
Deferencing a `nil` pointer leads to panic.

```suggestion
		if shallows == nil {
```

Reply:
This is a private function that we only call in 1 location, it will also contain a value (I would have passed by reference if go had that feature but it doesn't). We need to check whether the location that this variable is pointing to is nil or not. I could change this to `if shallows != nil && *shallows != nil {` but IMHO that would add an extra unnecessary check.